codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/message/MessageUnmarshallableGenerator.kt

```diff
@@ -31,122 +33,88 @@ class MessageUnmarshallableGenerator(val ctx: ProtocolGenerator.GenerationContex
             .build()
 
         val streamShape = ctx.model.expectShape<UnionShape>(streamingMember.target)
```

Comment:
As in the `MessageMarshallableGenerator` above, this takes the XML type and makes it universal for all protocols.